src/Settings.pm

```diff
@@ -428,14 +428,26 @@ sub loadByHandle {
 	assert(defined $object) if DEBUG;
 
 	my $filename;
-	if ($object->{autoSearch}) {
-		if ($object->{type} == CONTROL_FILE_TYPE) {
-			$filename = _findFileFromFolders($object->{name}, \@controlFolders);
+	my $internalFilename;
+	if ($object->{'autoSearch'}) {
```

Comment:
$internalFilename depends on $object->{autoSearch} because if it is '1' then $object->{name} holds the file name, if it is '0' then $object->{name} holds a file path and $object->{internalName} holds the file name. Even so your idea sure is better than just a bunch of ifs.